contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# This script only gets executed on systemd systems, see mount-zfs.sh for non-systemd systems
+
+# import the libs now that we know the pool imported
+[ -f /lib/dracut-lib.sh ] && dracutlib=/lib/dracut-lib.sh
+[ -f /usr/lib/dracut/modules.d/99base/dracut-lib.sh ] && dracutlib=/usr/lib/dracut/modules.d/99base/dracut-lib.sh
+. "$dracutlib"
+
+# load the kernel command line vars
+[ -z "$root" ] && root=$(getarg root=)
+# If root is not ZFS= or zfs: or rootfstype is not zfs then we are not supposed to handle it.
+[ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
+
+# There is a race between the zpool import and the pre-mount hooks, so we wait for a pool to be imported
+while true; do
+    [[ $(zpool list -H | grep -q -v '^$') -eq 0 ]] && break
+    [[ $(systemctl is-failed zfs-import-scan.service) == 'failed' ]] && exit 1
+    [[ $(systemctl is-failed zfs-import-cache.service) == 'failed' ]] && exit 1
```

Comment:
Can you instead check the new `zfs-import.target` rather than both of these or does it not report the state?

Reply:
will test and be in the next revision if it works

Reply:
nope, that target isn't found in the installed initram